resilience4j-rxjava2/src/main/java/io/github/resilience4j/bulkhead/operator/DisposableBulkhead.java

```diff
@@ -15,7 +17,7 @@
     private Disposable disposable;
     private final Bulkhead bulkhead;
     private final AtomicBoolean disposed = new AtomicBoolean(false);
-    private final AtomicBoolean permitted = new AtomicBoolean(false);
+    private final AtomicReference<Permit> permitted = new AtomicReference<>(Permit.PENDING);
```

Comment:
@danielimre could you please describe reasons behind such solution.

Reply:
Yes, I would also like to understand it :)

Reply:
It's really an edge-case and maybe an overkill but `acquireCallPermit` wasn't safe from this point of view. One subscriber can try to acquire a permit and fail which results in `permitted=false` and the rest of the subscription should act according to this. But meanwhile another subscriber could try to acquire a permit again and succeed which can have weird side-effects on the first subscription.
On the other hand, some parts of the operator not really thread-safe anyway so it's a question whether we should ensure this level of protection at all.

Reply:
As far as I know all operators in RxJava which operate on one Observable are NOT thread safe.
Only operators which operate over multiple Observables are thread safe: merge(), combineLatest(), zip(), .....

Reply:
So what should be done? Simplify it according to this?